src/Validator/PartyIdentificationValidator.php

```diff
@@ -25,12 +26,24 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        // Ensure identifier looks like a CPR or CVR number
-        if (!preg_match('/^(\d{6}-?\d{4}|\d{8})$/', $value->getIdentifier(), $matches)) {
-            $message = $this->translator->trans('The value { value } is not a valid CPR or CVR number.', ['{ value }' => $value->getIdentifier()], 'validator');
-            $this->context->buildViolation($message)
-                ->addViolation()
-            ;
+        $identifier = $value->getIdentifier();
+
+        if (IdentifierChoices::IDENTIFIER_TYPE_CHOICES['CPR'] === $value->getType()) {
```

Comment:
Consider using `IdentificationHelper::IDENTIFIER_TYPE_CPR` here.

And maybe use a `switch` statement, e.g.

```php
$type = $value->getType();
$identifier = $this->getIdentifier();
switch ($type) {
case IdentificationHelper::IDENTIFIER_TYPE_CPR:
  …
  break;

case IdentificationHelper::IDENTIFIER_TYPE_CVR:
  …
  break;

default:
  // Complain about invalid identifier type.
  break;
}
```